### PR TITLE
Run verification functions in task asynchronusly

### LINKED ISSUE #1321: Run verification functions in task asynchronusly
Check where LocalComputer is used in tasks (for creating data for verification, for verification process, for final result collecting) and make those methods asynchronous.

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #1321

### CODE REVIEW

**Review by ggruszczynski — APPROVED** [review]
ok, just need to merge new code & resolve conflicts.

**update**
Unforunately the tests fails.

File: apps/lux/task/verificator.py [hunk 1]
```diff
@@ -27,31 +28,35 @@ def __init__(self, *args, **kwargs):
...
+            ref_img_pil = load_as_PILImgRepr(paths.pop())
+            return ref_img_pil
 
-            ref_img_pil = load_as_PILImgRepr(f.pop())
-            ref_imgs.append(ref_img_pil)
+        with multiprocessing.Pool() as pool:
+            ref_imgs = pool.map(
+                f,
+                ((task, i) for i in range(task.reference_runs))
+            )
```
Comment (ggruszczynski) [inline apps/lux/task/verificator.py]:
``` python
 with multiprocessing.Pool() as pool:
             ref_imgs = pool.map(
                f,
                 ((task, i) for i in range(task.reference_runs))
             )
```

causes tests to fail:

```python
docker/test_docker_luxrender_task.py:261: in _test_luxrender_real_task
    result_type=ResultType.FILES)
../../apps/core/task/coretask.py:208: in computation_finished
    result_files, self)
../../golem/core/common.py:141: in func_wrapper
    return func(*args, **kwargs)
../../apps/core/task/verificator.py:50: in verify
    self._check_files(subtask_id, subtask_info, tr_files, task)
../../apps/lux/task/verificator.py:109: in _check_files
    ref_imgs = self._get_reference_imgs(task)
../../apps/lux/task/verificator.py:75: in _get_reference_imgs
    ((task, i) for i in range(task.reference_runs))
/usr/lib/python3.5/multiprocessing/pool.py:260: in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
/usr/lib/python3.5/multiprocessing/pool.py:608: in get
    raise self._value
/usr/lib/python3.5/multiprocessing/pool.py:385: in _handle_tasks
    put(task)
/usr/lib/python3.5/multiprocessing/connection.py:206: in send
    self._send_bytes(ForkingPickler.dumps(obj))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'multiprocessing.reduction.ForkingPickler'>
obj = (1, 0, <function mapstar at 0x7f02fb375620>, ((<function LuxRenderVerificator._get_reference_imgs.<locals>.f at 0x7f032c968378>, ((<Task: <Header: 'afb97625-8731-4492-9321-c19b6bb479b4'>>, 0),)),), {})
protocol = None

    @classmethod
    def dumps(cls, obj, protocol=None):
        buf = io.BytesIO()
>       cls(buf, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'LuxRenderVerificator._get_reference_imgs.<locals>.f'
```

**Review by ggruszczynski — CHANGES_REQUESTED** [review]

**Comment by jiivan** [comment]
@ggruszczynski tests fixed

**Review by mfranciszkiewicz — CHANGES_REQUESTED** [review]
This PR introduces concurrency but not asynchronicity. The executing thread is still blocked with `join`s. From my understanding, the point of #1321 is to convert aforementioned methods to coroutines.

**Review by ggruszczynski — APPROVED** [review]

**Comment by ggruszczynski** [comment]
As mfranciszkiewicz commented this task can be done better. 
I propose to merge this PR. Improvement can be done in another PR in a convenient moment.

**Comment by ggruszczynski** [comment]
in taskmanager.add_new_task(...), the task.create_reference_data_for_task_validation() shall be run asynchronously. Results are needed in class CoreVerificator(object) -->  def _check_files(self, subtask_id, subtask_info, tr_files, task):

**Comment by jiivan** [comment]
Verification changed. Needs a fresh start.

**Pull request closed without merging**